blank_rust_project/src/main.js

```diff
@@ -62,14 +62,29 @@ function signedInFlow() {
     // Displaying the signed in flow container.
     document.getElementById('signed-in-flow').classList.remove('d-none');
 
-    window.contract.welcome({account_id:window.accountId}).then(response => document.getElementById('speech').innerText = response.text);
+    getGreeting();
 
     // Adding an event to a sign-out button.
     document.getElementById('sign-out-button').addEventListener('click', () => {
         walletAccount.signOut();
         // Forcing redirect.
         window.location.replace(window.location.origin + window.location.pathname);
     });
+
+    // Adding an event to change greeting button.
+    document.getElementById('change-greeting').addEventListener('click', () => {
+        setGreeting();
+    });
+}
+
+async function setGreeting() {
+    await window.contract.set_greeting({message:'Howdy'});
+    getGreeting();
```

Comment:
Do we explain why this is necessary somewhere? I find it strange that you need to `await` two things in a row – write, then read – rather than having setter-functions return values from the blockchain. Why can't someone write a smart contract function that would allow better ergonomics here?

```js
async function setGreeting(message) {
  const { text } = await window.contract.set_greeting({ message })
  document.getElementById('speech').innerText = text
}
```

Reply:
This is different from setter-getter pattern (and we should probably not encourage setter-getter pattern anyways). I renamed the get fn to welcome to align better with smart contract function name.